apps/web-client/src/lib/forms/DaisyUIBookForm.svelte

```diff
@@ -0,0 +1,90 @@
+<script lang="ts">
+	import type { FormOptions, SuperValidated, SuperForm } from "sveltekit-superforms";
+	import { superForm, numberProxy, stringProxy } from "sveltekit-superforms/client";
+
+	import { testId } from "@librocco/shared";
+
+	import type { BookFormSchema } from "$lib/forms/schemas";
+
+	export let data: SuperValidated<BookFormSchema>;
+	export let options: FormOptions<BookFormSchema>;
+	export let publisherList: string[] = [];
+
+	export let isExtensionAvailable: boolean;
+
+	/**
+	 * Handle click of "X" icon button
+	 */
+	export let onCancel: (e: Event) => void = () => {};
+
+	export let onFetch: (isbn: string, form: SuperForm<BookFormSchema>["form"]) => void = () => {};
+
+	const form = superForm(data, options);
+
+	const { form: formStore, constraints, enhance } = form;
+
+	const priceProxy = numberProxy(formStore, "price", { empty: "undefined" });
+	const publisherProxy = stringProxy(formStore, "publisher", { empty: "undefined" });
+</script>
+
+<form
+	class="divide-y-gray-50 flex h-auto flex-col gap-y-6 px-4 py-4"
+	aria-label="Edit book details"
+	use:enhance
+	method="POST"
+	id={testId("book-form")}
+>
+	<div class="flex justify-between">
+		<label class="input-bordered input flex items-center gap-2">
+			ISBN
+			<input bind:value={$formStore.isbn} name="isbn" type="text" class="grow" placeholder="0000000000" {...$constraints.isbn} disabled />
+		</label>
+		<button disabled={!isExtensionAvailable} type="button" class="btn-secondary btn" on:click={() => onFetch($formStore.isbn, formStore)}>
+			Fill details
+		</button>
+	</div>
+
+	<label class="input-bordered input flex items-center gap-2">
+		Title*
+		<input bind:value={$formStore.title} name="title" type="text" class="grow" {...$constraints.title} required />
+	</label>
+	<label class="input-bordered input flex items-center gap-2">
+		Price*
+		<input bind:value={$priceProxy} class="grow" name="price" type="float" placeholder="0" required />
+		€
+	</label>
+	<label class="input-bordered input flex items-center gap-2">
+		Year
+		<input bind:value={$formStore.year} name="year" type="number" {...$constraints.year} />
+	</label>
+	<label class="input-bordered input flex items-center gap-2">
+		Authors
+		<input bind:value={$formStore.authors} name="authors" type="text" {...$constraints.authors} />
+	</label>
+	<label class="form-control w-full">
+		<div class="label">Publisher</div>
```

Comment:
@silviot @ikusteu on the inventory side, an employee should be able to type in a new publisher and save the form with this new option right? Should this be possibe here?

Reply:
I think so, yeah... (if I remember correctly, the publisher list is made by a query retrieving all distinct `publisher` values -- no able exists)

Reply:
@delasous Yes the inventory book form features a custom input component and some meltUI magic to provide a "select2" like functionality, thus we have a regression here.
I spent some time trying to re-implement this with DaisyUI in this contenxt but ultimately opted for a clean/easy select field since DaisyUI doesn't provide any kind of searchable select layout and it was taking me too much time to figure this out.
@ikusteu @delasous I suggest addressing this in a dedicated issue, what do you think?

Reply:
Yeah I agree. Just clicking through the old form and it seems to be a little buggy, so we could fix that in the new unit of work too